src/encord_active/lib/project/local.py

```diff
@@ -0,0 +1,125 @@
+import json
+from itertools import chain
+from pathlib import Path
+from typing import List, Optional, Union
+
+import yaml
+from encord.ontology import OntologyStructure
+from tqdm.auto import tqdm
+
+from encord_active.lib.encord.local_sdk import LocalUserClient
+from encord_active.lib.metrics.execute import run_metrics
+
+IMAGE_DATA_UNIT_FILENAME = "image_data_unit.json"
+
+
+class NoFilesFoundError(Exception):
+    """Exception raised when searching for files yielded an empty result."""
+
+    def __init__(self):
+        super().__init__("Couldn't find any files to import from the given specifications.")
+
+
+class ProjectExistsError(Exception):
+    """Exception raised when the target project already exists."""
+
+    def __init__(self, project_path: Path):
+        super().__init__(f"The project path `{project_path}` already exists.")
+        self.project_path = project_path
+
+
+def init_local_project(
+    root: Path,
+    target: Path,
+    glob: Optional[List[str]] = None,
+    project_name: str = "",
+    symlinks: bool = False,
+    dryrun: bool = False,
+) -> Union[Path, List[Path]]:
+    """
+    Initialising an Encord Active project based on the data found from the `root`
+    based on the `glob` arguments.
+
+    Args:
+        root: The root from which to search for files.
+        target: The directory in which the new project will be stored.
+        glob: A list of glob expressions that will be used to match image files.
+        project_name: A specific project name to use. If left out, the project
+            name will be the root directory name prepended with "[EA] "
+        symlinks: If false, files will be copied. If true, files will be symlinked.
+        dryrun: If false, a project will be created. If false, a list of all the
+            matched files will be returned.
+
+    Returns:
+        The path to the project directory.
+        If `dryrun` is True, a list of all the matched files will be returned
+        without actually initialising a project.
+    """
+    if not project_name:
+        project_name = f"[EA] {root.name}"
+
+    project_path = target / project_name
+
+    if project_path.is_dir():
+        raise ProjectExistsError(project_path)
+
+    if glob is None:
+        glob = ["**/*.jpg", "**/*.jpeg", "**/*.png", "**/*.tiff"]
+
+    files = list(chain(*[root.glob(g) for g in glob]))
+    if dryrun:
+        return files
+
+    if not len(files):
+        raise NoFilesFoundError()
```

Comment:
If people want to use the `init_local_project` via code, I thought that it would be nicer to return an empty list with the dryrun than raising an exception?
At least that would make their code prettier.
But we can swap the order.

Reply:
I personally hate it when I run something with a dry run flag and it isn't consistent with real runs. But I don't think it is a critical inconsistency, so up to you to decide how to go about this 🙂